### DISCUSSION TITLE
docker-compose errors

### DISCUSSION THREAD

**Discussion by raduganea** [EasyEngine/easyengine#1822] [Support Requests]
Hi guys, anyone facing docker-compose issues lately?

**This is my error from ee.log:**
Failed to execute script docker-compose  ERROR: for 0b2ffadff097_sitecom_nginx_1  'ContainerConfig'  ERROR: for nginx  'ContainerConfig' Traceback (most recent call last):   File "docker-compose", line 3, in <module>   File "compose/cli/main.py", line 81, in main   File "compose/cli/main.py", line 203, in perform_command   File "compose/metrics/decorator.py", line 18, in wrapper   File "compose/cli/main.py", line 1186, in up   File "compose/cli/main.py", line 1182, in up   File "compose/project.py", line 702, in up   File "compose/parallel.py", line 108, in parallel_execute   File "compose/p
arallel.py", line 206, in producer   File "compose/project.py", line 688, in do   File "compose/service.py", line 581, in execute_convergence_plan   File
"compose/service.py", line 503, in _execute_convergence_recreate   File "compose/parallel.py", line 108, in parallel_execute   File "compose/parallel.py", line 206, in producer   File "compose/service.py", line 496, in recreate   File "compose/service.py", line 615, in recreate_container   File "compose/ser
vice.py", line 334, in create_container   File "compose/service.py", line 922, in _get_container_create_options   File "compose/service.py", line 962, in
_build_container_volume_options   File "compose/service.py", line 1549, in merge_volume_bindings   File "compose/service.py", line 1579, in get_container_data_volumes KeyError: 'ContainerConfig'

**ee cli info**
OS                | Linux 5.15.0-91-generic #101-Ubuntu SMP Tue Nov 14 13:30:08 UTC 2023 x86_64 
Shell             | /bin/bash                                                                   
PHP binary        | /usr/bin/php8.0                                                             
PHP version       | 8.0.30                                                                      
php.ini used      | /etc/php/8.0/cli/php.ini                                                    
EE root dir       | phar://ee.phar                                                              
EE vendor dir     | phar://ee.phar/vendor                                                       
EE phar path      | /root                                                                       
EE packages dir   |                                                                             
EE global config  | /opt/easyengine/config/config.yml                                           
EE project config |                                                                             
EE version        | 4.6.6

**lsb_release -a**
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.4 LTS
Release:        22.04
Codename:       jammy

**docker version**
Client: Docker Engine - Community
 Version:           26.0.0
 API version:       1.45
 Go version:        go1.21.8
 Git commit:        2ae903e
 Built:             Wed Mar 20 15:17:48 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          26.0.0
  API version:      1.45 (minimum version 1.24)
  Go version:       go1.21.8
  Git commit:       8b79278
  Built:            Wed Mar 20 15:17:48 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.28
  GitCommit:        ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

**docker-compose version**
docker-compose version 1.29.2, build 5becea4c
docker-py version: 5.0.0
CPython version: 3.7.10
OpenSSL version: OpenSSL 1.1.0l  10 Sep 2019

**Comment by nelscon538** [comment]
I experience the same issue! even on a fresh server install i get docker-compose errors on Ubuntu server Ubuntu 22.04.4 LTS x86_64

https://github.com/EasyEngine/site-type-wp/issues/221

**Comment by me-2-you** [comment]
There may be new versions of Docker or Docker Compose... Try to roll back to previous versions..

With such versions, even on Ubuntu 20.04 they give the same error...

Additional — https://forums.docker.com/t/docker-compose-up-fails-after-yesterday-update-possible-regression-in-2-25-0-1/140430

**Comment by pagaling** [comment]
I'm also getting this error whenever I make a new Wordpress site. All DNS points correctly and SSL also works but when its time for Docker, it gives out this error

Starting site's services.
Warning: There was some error in docker-compose up.
Warning: Initiating clean-up.

And when I check the EE Logs, it says:

COMMAND: docker-compose -f docker-compose.yml  up -d nginx

STDERR: testsitecom_php_1 is up-to-date Recreating testsitecom_nginx_1 ...  [33272] Failed to execute script docker-compose

[15-04-2024 14:48:09] ee.DEBUG: RETURN CODE: 255

Is there any solution to this? I don't want to downgrade my Docker as it I am afraid that it may affect my server configuration.

**Comment by BuiThang** [comment]
I'm getting this error too.
I tested on both of Ubuntu 20.04 and Ubuntu 22.04, install new server and easyengine v4 but i cannot create site with ssl.
I tried with:
`ee site create domain.com --type=wp --ssl=le`
And tried with
ee site create domain.com --type=wp --ssl=le --le-mail=<email>
it's not working.
`Starting site's services.
Warning: There was some error in docker-compose up.
Warning: Initiating clean-up.
`

**Reply by pagaling** [reply]
I hope the EasyEngine guys notices this. It looks like an update is needed to be compatible with Docker Compose V2

**Reply by BuiThang** [reply]
Thank for reply. I fixed it by

> 1. go to your site: /opt/easyengine/sites/example.com
> 2.  and run `docker compose down` 
> 3. then `docker-compose -f docker-compose.yml  up -d nginx` and `docker compose up` 
> 4. then you can rerun the command with --ssl and it will work

**Reply by pagaling** [reply]
Thank you for this. I tried this but got an error on the step "docker compose up". Here are my steps

cd /opt/easyengine/sites/example.com
docker compose down 
docker-compose -f docker-compose.yml  up -d nginx
docker compose up 
ee site update example.com --ssl=le

My error:

WARN[0000] /opt/easyengine/sites/example.com/docker-compose.yml: `version` is obsolete
WARN[0000] volumes.htdocs: external.name is deprecated. Please set name and external: true
WARN[0000] volumes.config_nginx: external.name is deprecated. Please set name and external: true
WARN[0000] volumes.config_php: external.name is deprecated. Please set name and external: true
WARN[0000] volumes.log_nginx: external.name is deprecated. Please set name and external: true
WARN[0000] volumes.log_php: external.name is deprecated. Please set name and external: true
WARN[0000] volumes.newrelic_sock: external.name is deprecated. Please set name and external: true
WARN[0000] volumes.config_postfix: external.name is deprecated. Please set name and external: true
WARN[0000] volumes.data_postfix: external.name is deprecated. Please set name and external: true
WARN[0000] volumes.ssl_postfix: external.name is deprecated. Please set name and external: true
WARN[0000] networks.global-frontend-network: external.name is deprecated. Please set name and external: true
WARN[0000] networks.global-backend-network: external.name is deprecated. Please set name and external: true
network example.com was found but has incorrect label com.docker.compose.network set to "example.com"

Can you provide your steps again? TIA

**Reply by pagaling** [reply]
Finally made it worked. I only have to remove line 1 inside the file docker-compose.yml that says "version: '3'". The file is located under /opt/easyengine/sites/example.com 

As soon as I removed that I proceeded to redo the steps:

1. cd /opt/easyengine/sites/example.com
2. docker compose down
3. docker-compose -f docker-compose.yml up -d nginx
4. docker compose up
5. ee site update example.com --ssl=le

I still got the errors after executing docker compose up but the site worked afterwards regardless of the errors

If I were to do the 1st time, these should be my steps:

1. create the site first without using --ssl=le (in my case, here is my command: ee site create example.com --type=wp --php=8.1 --cache --admin-email=mail@example.com --admin-user=username --admin-pass=userpassword)
2. cd /opt/easyengine/sites/example.com
3. docker compose down
4. docker-compose -f docker-compose.yml up -d nginx
5. docker compose up
6. Then update the site by using this command: ee site update example.com --ssl=le

Hope this helps
